src/3d/materials/qgsgoochmaterialsettings.cpp

```diff
@@ -148,10 +148,10 @@ void QgsGoochMaterialSettings::addParametersToEffect( Qt3DRender::QEffect * ) co
 QByteArray QgsGoochMaterialSettings::dataDefinedVertexColorsAsByte( const QgsExpressionContext &expressionContext ) const
 {
 
-  const QColor diffuse = dataDefinedProperties().valueAsColor( Diffuse, expressionContext, mDiffuse );
-  const QColor warm = dataDefinedProperties().valueAsColor( Warm, expressionContext, mWarm );
-  const QColor cool = dataDefinedProperties().valueAsColor( Cool, expressionContext, mCool );
-  const QColor specular = dataDefinedProperties().valueAsColor( Specular, expressionContext, mSpecular );
+  const QColor diffuse = dataDefinedProperties().valueAsColor( static_cast< int >( QgsAbstractMaterialSettings::Property::Diffuse ), expressionContext, mDiffuse );
```

Comment:
all those explicit casts to int are quite ugly. Isn't there some template based solution that could automatically do it.

like 
```
template<class T> QColor valueAsColor( const T& key, const QgsExpressionContext &context, const QColor &defaultColor = QColor(), bool *ok = nullptr ) const
{
    return valueAsColor(static_cast<int>(key), context, defaultColor, ok);
}
```

Reply:
Oh yeah, that's MUCH cleaner! :100: